public/javascripts/rateRecipe.js

```diff
@@ -1,3 +1,90 @@
 document.addEventListener('DOMContentLoaded', () => {
-    
+  const getRecipeIdAndStarRating = (element) => {
+    const ids = element.id.split('.');
+    const recipeId = parseInt(ids[0].split(':')[1], 10);
+    const starRating = parseInt(ids[1].split(':')[1], 10);
+
+    return { recipeId, starRating };
+  };
+
+  const anyIsChecked = (recipeId, num) => {
+    for (let i = 1; i < num; i++) {
+      const star = document.getElementById(`recipe:${recipeId}.star:${i}`);
+      if (star.classList.contains('checked')) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  const fillStars = (recipeId, starRating) => {
+    for (let i = 1; i <= starRating; i++) {
+      const star = document.getElementById(`recipe:${recipeId}.star:${i}`);
+      star.classList.add('checked');
+    }
+    for (let i = starRating + 1; i <= 5; i++) {
+      const star = document.getElementById(`recipe:${recipeId}.star:${i}`);
+      star.classList.remove('checked');
+    }
+  };
+
+  document.querySelectorAll('.rating__button').forEach((button) => {
+    button.addEventListener('click', async (event) => {
+      const { recipeId, starRating } = getRecipeIdAndStarRating(event.target);
+      if (event.target.classList.contains('checked') || anyIsChecked(recipeId, starRating)) {
+        const res = await updateRateRecipe(recipeId, starRating);
+        if (res) {
+          fillStars(recipeId, starRating);
+        } else {
+          alert('Something went wrong. Please try again');
+        }
+      } else {
+        const res = await rateRecipe(recipeId, starRating);
+        if (res) {
+          //fill in corresponding star
```

Comment:
If you would like to leave comments in your code, try putting a single line summary of a larger block of code. Individual comments break up the flow and can make it harder to read the code instead of easier.